e2e/framework/utils/k8s_helper.go

```diff
@@ -304,3 +286,45 @@ func (k8sh *K8sHelper) GetPodHostId(podNamePattern string, namespace string) (st
 		return serr, fmt.Errorf("Error Getting Monitor IP")
 	}
 }
+
+func (k8sh *K8sHelper) IsStorageClassPresent(name string) (bool, error) {
+	cmdArgs := []string{"get", "storageclass", "-o", "jsonpath='{.items[*].metadata.name}'"}
+	sout, serr, _ := ExecuteCmd("kubectl", cmdArgs)
+	if strings.Contains(sout, name) {
+		return true, nil
+	}
+	return false, fmt.Errorf("Storageclass %s not found, err ->%s", name, serr)
+
+}
+
+func (k8sh *K8sHelper) GetPVCStatus(name string) (string, error) {
+	cmdArgs := []string{"get", "pvc", "-o", "jsonpath='{.items[*].metadata.name}'"}
+	sout, serr, _ := ExecuteCmd("kubectl", cmdArgs)
+	if strings.Contains(sout, name) {
+		cmdArgs := []string{"get", "pvc", name, "-o", "jsonpath='{.status.phase}'"}
+		res, _, _ := ExecuteCmd("kubectl", cmdArgs)
+		return res, nil
+	}
+	return "PVC NOT FOUND", fmt.Errorf("PVC %s not found,err->%s", name, serr)
+}
+
+func (k8sh *K8sHelper) IsPodInExpectedState(podNamePattern string, namespace string, state string) bool {
+	cmdArgs := []string{"get", "pods", "-l", "app=" + podNamePattern, "-o", "jsonpath={.items[0].status.phase}", "--no-headers=true"}
+	if namespace != "" {
+		cmdArgs = append(cmdArgs, []string{"-n", namespace}...)
+	}
+	inc := 0
+	for inc < 10 {
+		res, _, status := ExecuteCmd("kubectl", cmdArgs)
+		if status == 0 {
+			if res == state {
+				return true
+			}
+		}
+		inc++
+		time.Sleep(10 * time.Second)
```

Comment:
seems like 10 seconds is a long wait time between checks, ill leave it up to you